app/kumactl/pkg/install/k8s/client_config.go

```diff
@@ -7,9 +7,14 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-func DefaultClientConfig() (*rest.Config, error) {
+func DefaultClientConfig(kubeconfig, context string) (*rest.Config, error) {
```

Comment:
Two string args is easy to make mistake with. Suggest adoption functional options (either here or in a followup PR).
```
k8s.DefaultClientConfig(KubeConfigOpt("foo"), KubeContextOpt("bar"))
```

I'd almost suggest using the controller-runtime API, except it will inject the `-kubeconfig` flag globally (invisible to cobra).

https://github.com/kubernetes-sigs/controller-runtime/blob/v0.9.2/pkg/client/config/config.go#L79